Implementation of MK docs. (#33)
FAIRmat-NFDI/nyaml: docs/index.md

# FAIRmat NYAML documentation

<div markdown="block" class="home-grid">
<div markdown="block">

### Tutorial

Tutorials to write different parts and a full NeXus application or base class

> - [Tutorials to write NeXus definition in YAML](tutorials.md)

Reviewer: ```suggestion
- [Tutorials for writing NeXus definition in YAML](tutorials.md)
```

</div>
<div markdown="block">

### How-to guides

How-to install and use the nyaml tool

- [How-to guide for NYAML tools](how_to_guide.md)

</div>

<div markdown="block">

### Learn

An introduction to NeXus and its design principles.

- [An introduction to NeXus](https://manual.nexusformat.org/index.html)

> - [Motivation for NYAML tool](explanations.md)

Reviewer: ```suggestion
- [Explanation of the nyaml tool](explanations.md)
```

</div>
<div markdown="block">

> ### Reference
> Relavent references that supported to develop this tool:
> - [NYAML GitHub repository](https://github.com/FAIRmat-NFDI/nyaml)
> - [FAIRmat-NeXus website](https://fairmat-experimental.github.io/nexus-fairmat-proposal/)
> - [FAIRmat-NeXus GitHub repository](https://github.com/FAIRmat-NFDI/nexus_definitions)
> - [Official werbsite for NeXus](https://manual.nexusformat.org/index.html)
> 
> - [NYAML developed under project FAIRmat-NFDI](https://github.com/FAIRmat-NFDI)
> - [National Research Data Infrastructure (NFDI) funded FAIRmat-NFDI](https://www.nfdi.de/?lang=en)
> </div>
> </div>

Reviewer: ```suggestion
### References

- [NYAML GitHub repository](https://github.com/FAIRmat-NFDI/nyaml)
- [FAIRmat-NeXus website](https://fairmat-experimental.github.io/nexus-fairmat-proposal/)
- [FAIRmat-NeXus GitHub repository](https://github.com/FAIRmat-NFDI/nexus_definitions)
- [Official NeXus website](https://manual.nexusformat.org/index.html)
</div>
</div>

<h2>Project and community</h2>

Any questions or suggestions? [Get in touch!](https://www.fair-di.eu/fairmat/about-fairmat/team-fairmat)

[The work is funded by the Deutsche Forschungsgemeinschaft (DFG, German Research Foundation) - 460197019 (FAIRmat).](https://gepris.dfg.de/gepris/projekt/460197019?language=en)
```

> # FAIRmat nyaml documentation

Reviewer 2: Wouldnt mention FAIRmat here but in the footer where indeed we should acknowledge the funding from the DFG

See an example here:
https://fairmat-nfdi.github.io/pynxtools-em/
bottom of the page with hyperlink to DFG page

Reviewer 2: Maybe also a statement supported by the NIAC is useful

Author: Addressed!

> The `nyaml` tool offers a user-friendly command-line interface for converting between `YAML` and `XML` representations of NeXus definitions written in the NeXus Definition Language (NXDL). While the official NeXus definitions are maintained in `XML,` authoring them in `YAML` provides a more concise and readable alternative. `nyaml` bridges this gap, allowing developers to work in `YAML` while ensuring compatibility with the official `XML` format.

Reviewer 2: swop order `XML` and `YAML` representations

Author: Addressed!

> The tool supports both application definitions and base classes. It can be installed via `pip` and includes comprehensive documentation and tutorials to assist users in writing NeXus definitions using `YAML`.

Reviewer 2: "The tool supports both application definitions and base classes." why to mention it when it supports all classes? Could left people wondering if contributed_definitions are not supported although they are.

Author: improved!

<div markdown="block" class="home-grid">
<div markdown="block">

### Tutorial

Tutorials to write different parts and a full NeXus application or base class

- [Writing a NeXus definition in YAML](tutorials/tutorials.md)

</div>
<div markdown="block">

### How-to guides

How-to install and use the nyaml tool

- [Installation and use of the `nyaml` tool](howtos/howto.md)

</div>

<div markdown="block">

### Learn

An introduction to NeXus and its use in the light of `YAML` format using the `nyaml` tool.

- [An introduction to NeXus](https://manual.nexusformat.org/index.html)
- [Motivation behind the nyaml tool](learn/explanations.md)

> - [Workflow of the nyaml tool with examples](learn/workflow.md)

Reviewer: These must be added to the mkdocs.yaml file in the `nav` section

Author: addressed!

# FAIRmat nyaml documentation

The `nyaml` tool is designed to be user-friendly, with a command line interface that allows users to easily inter-convert between `YAML` and `XML` formats written in terms of NXDL (NeXus Definition Language). This toolset is particularly beneficial for scientists and developers who are familiar with the NeXus format and wish to leverage the advantages of `YAML` for schema development. The tool supports the conversion of both application definitions and base classes.

> The tool can be installed via pip, and it provides comprehensive documentation and tutorials to assist users in writing NeXus definitions in `YAML`.

Reviewer: ```suggestion
The `nyaml` tool offers a user-friendly command-line interface for converting between `YAML` and `XML` representations of NeXus definitions written in the NeXus Definition Language (NXDL). While the official NeXus definitions are maintained in `XML,` authoring them in `YAML` provides a more concise and readable alternative. `nyaml` bridges this gap, allowing developers to work in `YAML` while ensuring compatibility with the official `XML` format.

The tool supports both application definitions and base classes. It can be installed via `pip` and includes comprehensive documentation and tutorials to assist users in writing NeXus definitions using `YAML`.
```

We should mention what the package solves at the beginning.

Author: Addressed!

> The NeXus data format, described by the NeXus Definition Language (NXDL), represents a concerted effort aimed at facilitating data exchange within scientific communities, particularly among those engaged in neutron, X-ray, and muon research [J. Appl. Cryst. (2015). 48, 301-305](https://doi.org/10.1107/S1600576714027575). The data format is also being used by the material science community under the project [NeXus-FAIRmat](https://fairmat-experimental.github.io/nexus-fairmat-proposal/) supporting FAIR (Findable, Accessible, Interoperable and Reuseable) data principles. It serves as a standardized framework for both data exchange and storage. At its core, the NeXus Definition Language (NXDL) functions as the cornerstone through which scientists delineate the nomenclature and organizational structure of information within NeXus data files, tailored to specific scientific techniques.

Reviewer: Now that I see it here, maybe this is actually too much for the general index. If we build the docs, we have to scroll a lot to see the different tiles for the individual parts of the plugin. My bad!

My suggestion:
- for the index page, go with a smaller description of the tool (like we do in the NOMAD [docs](https://nomad-lab.eu/prod/v1/staging/docs/))
- Whatever does fit into this short introduction, move it back to the explanations.
- For this, I suggest that we have two pages in the explanations part:
  - One is an actual explanation of what the tool is for (i.e., the long text we now have in index.md) -> goes to explanations.md
  - The other is what we already have in the explanations section. This is concerned with the workflow, so I would move it to a new file called workflow.md
- Of course, the links in index.md and mkdocs.yml need to be updated afterwards.

Sounds okay?

Author: Addressed at 53d263ed5edd4ec3ca20fb55b07674980f8d137e

> The NeXus data format, described by the NeXus Definition Language (NXDL), represents a concerted effort aimed at facilitating data exchange within scientific communities, particularly among those engaged in neutron, X-ray, and muon research [J. Appl. Cryst. (2015). 48, 301-305](https://doi.org/10.1107/S1600576714027575). The data format is also being used by the material science community under the project [NeXus-FAIRmat](https://fairmat-experimental.github.io/nexus-fairmat-proposal/) supporting FAIR (Findable, Accessible, Interoperable and Reuseable) data principles. It serves as a standardized framework for both data exchange and storage. At its core, the NeXus Definition Language (NXDL) functions as the cornerstone through which scientists delineate the nomenclature and organizational structure of information within NeXus data files, tailored to specific scientific techniques.

Reviewer 2: "what does the "It" in "data principles. It" refer to?

Author: Addresed!

<div markdown="block" class="home-grid">
<div markdown="block">

### Tutorial

Tutorials to write different parts and a full NeXus application or base class

- [Writing a NeXus definition in YAML](tutorials.md)

</div>
<div markdown="block">

### How-to guides

How-to install and use the nyaml tool

> - [Use of `nyaml` tool](how_to_guide.md)

Reviewer: ```suggestion
- [Installation and use of the `nyaml` tool](how_to_guide.md)
```

Author: Addressed!

</div>

<div markdown="block">

### Learn

> An introduction to NeXus and its design principles.

Reviewer: Remove this sentence, it does not match the individual pages

Author: Modified!

> NXDL is used to define general data storage objects (base classes) and the base classes are the building blocks for defining measurement-specific or even instrument-specific or software-specific data storage objects (application definitions). In this process, members and definitions of individual base classes can be used as is or customized. In essence, the process of schema development, whether for a base class or an application definition, entails crafting an NXDL schema definition file with the extension 'nxdl.xml', utilizing the Extensible Markup Language, [XML](https://www.w3.org/TR/REC-xml/REC-xml-20081126.xml).

Reviewer 2: Adding to Lukas' recommendation I suggest to do the splitting first and then start a revisioning of the text paragraph snippets remaining, for this reason I skip the revision here for this file.

> - [Tutorials for writing NeXus definition in YAML](tutorials.md)

Reviewer: ```suggestion
- [Writing a NeXus definition in YAML](tutorials.md)
```

> - [How-to guide for nyaml tools](how_to_guide.md)

Reviewer: ```suggestion
- [How-to guide for using the `nyaml` tools(how_to_guide.md)
```

### Learn

An introduction to NeXus and its design principles.

- [An introduction to NeXus](https://manual.nexusformat.org/index.html)
- [Explanation of the nyaml tool](explanations.md)

> 

Reviewer: Add a third page here for the workflow, as described above
